core/src/main/scala/org/apache/spark/sql/delta/ConflictChecker.scala

```diff
@@ -231,16 +232,17 @@ private[delta] class ConflictChecker(
       // Fail if files have been deleted that the txn read.
       val readFilePaths = currentTransactionInfo.readFiles.map(
         f => f.path -> f.partitionValues).toMap
-      val deleteReadOverlap = winningCommitSummary.removedFiles
+      val deleteReadOverlap = winningCommitSummary.changedDataRemovedFiles
```

Comment:
update the scaladoc for this method. we aren't just checking RemoveFile actions. we are checking RemoveFIle actions that actually change data (e.g. ignore OPTIMIZE)